### PR TITLE
Update Kuberentes doc to 1.5 (StatefulSets)

### LINKED ISSUE #932: Update kubernetes docs for 1.5
In kubernetes 1.5, PetSets have been renamed to StatefulSets; we should update our docs accordingly.

http://kubernetes.io/docs/tasks/manage-stateful-set/upgrade-pet-set-to-stateful-set/

### PR DESCRIPTION
**Pull request by jseldess**
Fixes #932 

Merge needs to wait on https://github.com/cockroachdb/cockroach/pull/12536

---
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/940)

### CODE REVIEW

File: orchestrate-cockroachdb-with-kubernetes.md [hunk 1]
```diff
@@ -4,9 +4,9 @@ summary: How to orchestrate the deployment and management of an insecure 3-node
...
+This page shows you how to orchestrate the deployment and management of an insecure 3-node CockroachDB cluster with [Kubernetes](http://kubernetes.io/), using the beta [StatefulSet](http://kubernetes.io/docs/concepts/abstractions/controllers/statefulsets/) feature.
 
-{{site.data.alerts.callout_info}}Running a stateful application such as CockroachDB on Kubernetes requires using some of Kubernetes' more complex features available only in alpha versions. There are easier ways to run CockroachDB on Kubernetes for testing purposes, but the method presented here is destined to become a production deployment once Kubernetes matures sufficiently.{{site.data.alerts.end}}
+{{site.data.alerts.callout_info}}Running a stateful application such as CockroachDB on Kubernetes requires using some of Kubernetes' more complex features available only in beta versions. There are easier ways to run CockroachDB on Kubernetes for testing purposes, but the method presented here is destined to become a production deployment once Kubernetes matures sufficiently.{{site.data.alerts.end}}
```
Comment (a-robinson) [inline orchestrate-cockroachdb-with-kubernetes.md]: The "requires using some of Kubernetes' more complex features available only in beta versions" line isn't quite as correct now that petsets/statefulsets are in beta. Beta features are available in all kubernetes deployments, whereas alpha features are only available in kubernetes deployments that specifically allow them.

A better phrasing might be something more like "requires using some of Kubernetes' more complex features that are at a beta level of support". We could perhaps link to what they mean by beta - http://kubernetes.io/docs/api/#api-versioning

File: orchestrate-cockroachdb-with-kubernetes.md [hunk 2]
```diff
@@ -40,7 +40,7 @@ Feature | Description
...
 [pod](http://kubernetes.io/docs/user-guide/pods/) | A pod is a group of one of more Docker containers. In this tutorial, each pod will run on a separate instance and contain one Docker container running a single CockroachDB node. You'll start with 3 pods and grow to 4.
-[PetSet](http://kubernetes.io/docs/user-guide/petset/) | A PetSet is a group of pods treated as stateful units, where each pod has distinguishable network identity and always binds back to the same persistent storage on restart. PetSets are an alpha feature as of Kubernetes version 1.4. As such, they may not be available in some production-ready hosting environments.
+[StatefulSet](http://kubernetes.io/docs/concepts/abstractions/controllers/statefulsets/) | A StatefulSet is a group of pods treated as stateful units, where each pod has distinguishable network identity and always binds back to the same persistent storage on restart. StatefulSets are an beta feature as of Kubernetes version 1.5. As such, they may not be available in some production-ready hosting environments.
```
Comment (a-robinson) [inline orchestrate-cockroachdb-with-kubernetes.md]: s/an beta/a beta/

And remove the part about not being available. Beta features are available in all kubernetes clusters.

File: orchestrate-cockroachdb-with-kubernetes.md [hunk 3]
```diff
@@ -51,7 +51,7 @@ Feature | Description
...
```
Comment (a-robinson) [inline orchestrate-cockroachdb-with-kubernetes.md]: ditto

Comment (jseldess) [inline orchestrate-cockroachdb-with-kubernetes.md]: Done

File: orchestrate-cockroachdb-with-kubernetes.md (hunk 2 above)
Comment (jseldess) [inline orchestrate-cockroachdb-with-kubernetes.md]: Done.

File: orchestrate-cockroachdb-with-kubernetes.md (hunk 1 above)
Comment (jseldess) [inline orchestrate-cockroachdb-with-kubernetes.md]: Done.

**Pull request merged**